pkg/crc/machine/vsock.go

```diff
@@ -0,0 +1,58 @@
+package machine
+
+import (
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/daemonclient"
+	"github.com/code-ready/gvisor-tap-vsock/pkg/types"
+	"github.com/pkg/errors"
+)
+
+func exposePorts() error {
+	daemonClient := daemonclient.New()
+	requestedPorts := vsockPorts()
+	alreadyOpenedPorts, err := daemonClient.List()
+	if err != nil {
+		return err
+	}
+	var missingPorts []types.ExposeRequest
+	for _, port := range requestedPorts {
+		if !isOpened(alreadyOpenedPorts, port) {
+			missingPorts = append(missingPorts, port)
+		}
+	}
+	for i := range missingPorts {
+		port := &requestedPorts[i]
```

Comment:
???

as you are wrote this it looks like you are poking into requestedPorts with the indexer of the missingPorts? It would be better to call this `for p := ` as this is the integer value of the actual port?

Reply:
yes good point! oups